server/src/main/java/org/opensearch/identity/ExtensionTokenProcessor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+
+/**
+ * Token processor class to handle token encryption/decryption
+ */
+public class ExtensionTokenProcessor {
+    private String extensionUniqueId;
+
+    public ExtensionTokenProcessor(String extensionUniqueId) {
+        this.extensionUniqueId = extensionUniqueId;
+    }
+
+    public String getExtensionUniqueId() {
+        return extensionUniqueId;
+    }
+
+    /**
+     * ENCRYPTION
+     *
+     * Create a two-way encrypted access token for given principal for this extension
+     * @return token generated from principal
+     */
+    public PrincipalIdentifierToken generateToken(Principal principal) {
+        String token = principal.getName() + ":" + extensionUniqueId;
+        return new PrincipalIdentifierToken(token);
+    }
+
+    /**
+     * DECRYPTION
+     *
+     * Convert token to Principal
+     * @return Principal
+     */
+    public Principal extractPrincipal(PrincipalIdentifierToken token) {
+        String[] parts = token.getToken().split(":");
+
+        if (parts.length != 2) {
+            return null;
+        }
+
+        return new Principal() {
+            public String getName() {
```

Comment:
Does this need `@Override` annotation?  Will it still have access to `parts` when this method finishes executing?  I'm not confident this will work as intended (a test showing that it does would help)

Reply:
I'm unsure whether it would need an Override annotation, conventionally it should. However the tests passed with/without the annotation.

Reply:
I have changed it to a lambda. `() -> parts[0];`

Reply:
Java might do the right thing, but the clearest way to resolve this is assign the value to a final local variable that is passed to the class.
```
final String name = parts[0];
return new Principal() {
    public String getName() {
        return name;
    }
}
```